index-handlers/src/main/java/no/sikt/nva/nvi/index/aws/OpenSearchIndexClient.java

```diff
@@ -0,0 +1,82 @@
+package no.sikt.nva.nvi.index.aws;
+
+import static com.amazonaws.auth.internal.SignerConstants.AUTHORIZATION;
+import static nva.commons.core.attempt.Try.attempt;
+import java.io.IOException;
+import no.sikt.nva.nvi.common.IndexClient;
+import no.sikt.nva.nvi.index.model.NviCandidateIndexDocument;
+import no.unit.nva.auth.CachedJwtProvider;
+import nva.commons.core.JacocoGenerated;
+import org.opensearch.client.opensearch.OpenSearchClient;
+import org.opensearch.client.opensearch._types.OpenSearchException;
+import org.opensearch.client.opensearch.core.IndexRequest;
+import org.opensearch.client.opensearch.indices.CreateIndexRequest;
+import org.opensearch.client.opensearch.indices.ExistsRequest;
+import org.opensearch.client.transport.aws.AwsSdk2Transport;
+import org.opensearch.client.transport.aws.AwsSdk2TransportOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.http.apache.ApacheHttpClient;
+import software.amazon.awssdk.regions.Region;
+
+@JacocoGenerated
+//TODO: Handle test coverage
+public class OpenSearchIndexClient implements IndexClient<NviCandidateIndexDocument> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OpenSearchIndexClient.class);
+    private static final String INDEX = "nviCandidates";
+    private final OpenSearchClient openSearchClient;
+
+    public OpenSearchIndexClient(String openSearchEndpoint, CachedJwtProvider cachedJwtProvider, Region region)
+        throws IOException {
+        this.openSearchClient = new OpenSearchClient(
+            new AwsSdk2Transport(
+                ApacheHttpClient.builder().build(),
+                openSearchEndpoint,
+                region,
+                transportOptionWithToken("Bearer " + cachedJwtProvider.getValue().getToken())
```

Comment:
The token you are providing will only last 15mins, but if your lambda is called repeatedly it can survive longer and the token will be invalid.
Instead you need to refresh the token on each call if its outdated. See search-api

Reply:
As discussed, solving in new PR